lib/pure/algorithm.nim

```diff
@@ -9,20 +9,36 @@
 
 ## This module implements some common generic algorithms.
 ##
+## Basic usage
+## ===========
+##
 ## .. code-block::
 ##    import algorithm
 ##
-##    var birth = ["2000 John", "1995 Marie", "2010 Jane"]
-##    sort(birth)
-##    assert birth == @["1995 Marie", "2000 John", "2010 Jane"]
-##    echo "Firstborn: " & birth[0] # --> 1995 Marie
+##    type People = tuple
+##    year: int
+##    name: string
+##
+##    var a: seq[People]
+##
+##    a.add((2000, "John"))
+##    a.add((2005, "Marie"))
+##    a.add((2010, "Jane"))
+##
+##    # Sorting with default system.cmp
+##    a.sort()
+##    assert a == @[(year: 2000, name: "John"), (year: 2005, name: "Marie"), (year: 2010, name: "Jane")]
+##
+##    proc myCmp(x, y: People): int =
+##      if x.name < y.name: -1 else: 1
 ##
-##    let john = binarySearch(birth, "2000 John")
-##    echo "John was born as number " & $(john + 1) # --> 2
+##    # Sorting with custom proc
+##    a.sort(myCmp)
+##    assert a == @[(year: 2010, name: "Jane"), (year: 2000, name: "John"), (year: 2005, name: "Marie")]
 ##
 ##
-## See also:
-## =========
+## See also
+## ========
 ## * `tables module<tables.html>`_ for sorting tables
```

Comment:
I think that link to `sequtils` might be of interest too.

Reply:
Ok? acd0c5635934bcf5d5bf0f3231abd26e7fab5a66